hardhat.config.ts

```diff
@@ -1,12 +1,97 @@
 import { HardhatUserConfig } from "hardhat/config";
 import "@nomicfoundation/hardhat-toolbox";
 import "hardhat-gas-reporter";
+import dotenv from "dotenv";
+import yargs from "yargs";
+import { HttpNetworkUserConfig } from "hardhat/types";
+import "hardhat-deploy";
+import { DeterministicDeploymentInfo } from "hardhat-deploy/dist/types";
+import { getSingletonFactoryInfo } from "@gnosis.pm/safe-singleton-factory";
+import { ethers } from "ethers";
+
+// Load environment variables.
+dotenv.config();
+
+const argv : any = yargs
+    .option("network", {
+        type: "string",
+        default: "hardhat",
+    })
+    .help(false)
+    .version(false).argv;
+
+const { NODE_URL, MNEMONIC, INFURA_KEY, ETHERSCAN_API_KEY} = process.env;
+
+const deterministicDeployment = (network: string): DeterministicDeploymentInfo => {
+  const info = getSingletonFactoryInfo(parseInt(network));
+  if (!info) {
+      throw new Error(`
+      Safe factory not found for network ${network}. You can request a new deployment at https://github.com/safe-global/safe-singleton-factory.
+      For more information, see https://github.com/safe-global/safe-contracts#replay-protection-eip-155
+    `);
+  }
+  return {
+      factory: info.address,
+      deployer: info.signerAddress,
+      funding: (ethers.toBigInt(info.gasLimit) * (ethers.toBigInt(info.gasPrice))).toString(),
+      signedTx: info.transaction,
+  };
+};
+
+
+if (["goerli", "mumbai"].includes(argv.network) && INFURA_KEY === undefined) {
+  throw new Error(`Could not find NODE_URL in env, unable to connect to network ${argv.network}`);
+}
+
+const sharedNetworkConfig: HttpNetworkUserConfig = {};
+
+sharedNetworkConfig.accounts = {
+  mnemonic: MNEMONIC || ""
+}
 
 const config: HardhatUserConfig = {
   solidity: "0.8.18",
```

Comment:
nit: 0.8.20 is the latest version

Reply:
Actully, hardhat gives warning with `0.8.19` and `0.8.20`. So, fixed the version to `0.8.18` which is officially supported by hardhat.
![Screenshot 2023-07-12 at 15 16 29](https://github.com/5afe/safe-protocol/assets/16358982/ec94b6db-10f8-4b97-8d54-e99fa855391b)

Reply:
In my experience, it still works fine and only affects development features like stack traces (mentioned in the error message). In case we experience errors during the development process, we could use 8.18 for the development